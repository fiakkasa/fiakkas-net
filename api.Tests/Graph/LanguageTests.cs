using HotChocolate.Execution;

namespace api.Tests.Graph;

public class LanguageTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task Languages_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              languages {
                totalCount
                nodes {
                  createdAt
                  id
                  internalId
                  proficiency
                  title
                  updatedAt
                  version
                }
              }
            }
            """);

        var fn = result.ExpectOperationResult;

        Assert.Null(fn().Errors);
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task LanguageById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              languageById(id: "TGFuZ3VhZ2UKZzAyYTNiZTliM2YwNDRiNGE4OTQ1ZTg0ZmVmNTM3YjU4") {
                createdAt
                id
                internalId
                proficiency
                title
                updatedAt
                version
              }
            }
            """);

        var fn = result.ExpectOperationResult;

        Assert.Null(fn().Errors);
        result.ToJson().MatchSnapshot();
    }
}
