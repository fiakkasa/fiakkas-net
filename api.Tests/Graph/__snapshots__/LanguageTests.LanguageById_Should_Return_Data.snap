{
  "data": {
    "languageById": {
      "createdAt": "2024-01-01T00:00:00.000Z",
      "id": "TGFuZ3VhZ2UKZzAyYTNiZTliM2YwNDRiNGE4OTQ1ZTg0ZmVmNTM3YjU4",
      "internalId": "02a3be9b-3f04-4b4a-8945-e84fef537b58",
      "proficiency": "NATIVE",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
